### PR TITLE
feat(payment): PAYPAL-2758 PayLater Messages

### PR DESCRIPTION
**Pull request by bc-nick**
## What?

Added `PayPalSDK` initialization option to `braintree-paypal-payment-strategy`

## Why?

To be able initialize Pay Later Messages from payment strategy

Related PR: https://github.com/bigcommerce/checkout-js/pull/1395

## Testing / Proof

Screenshot 2023-08-01 at 13 58 03

All tests passed

@bigcommerce/team-checkout @bigcommerce/team-payments

### CODE REVIEW

File: packages/core/src/payment/payment-methods.mock.ts [hunk 1]
```diff
@@ -55,6 +55,7 @@ export function getBraintreePaypal(): PaymentMethod {
         clientToken: 'foo',
         initializationData: {
             isBrainteeVenmoEnabled: false,
+            isPayLaterBannersEnabled: false,
```
Comment (bc-nick) [inline packages/core/src/payment/payment-methods.mock.ts:58]: The temporary name, will probably have a different name

**Review by vitalii-koshovyi — APPROVED** [review]

File: packages/core/src/payment/strategies/braintree/braintree-paypal-payment-strategy.ts [hunk 2]
```diff
@@ -47,6 +52,10 @@ export default class BraintreePaypalPaymentStrategy implements PaymentStrategy {
                 .paymentMethods.getPaymentMethodOrThrow(methodId);
         }
 
+        if (this._paymentMethod.clientToken && braintreeOptions?.bannerContainerId) {
+            return this._loadPaypalCheckoutInstance(braintreeOptions);
```
Comment (serhii-tkachenko) [inline packages/core/src/payment/strategies/braintree/braintree-paypal-payment-strategy.ts:56]: Should we load PayPal as on line # 60 as well?

File: packages/core/src/payment/strategies/braintree/braintree-paypal-payment-strategy.spec.ts
Comment (serhii-tkachenko) [inline packages/core/src/payment/strategies/braintree/braintree-paypal-payment-strategy.spec.ts:1]: May I ask you to add more test for banner rendering implementation?

Comment (bc-nick) [reply packages/core/src/payment/strategies/braintree/braintree-paypal-payment-strategy.spec.ts:1]: Added test for the banner rendering implementation

**Review by serhii-tkachenko — APPROVED** [review]

File: packages/core/src/payment/strategies/braintree/braintree-paypal-payment-strategy.ts [hunk 3]
```diff
@@ -223,6 +238,71 @@ export default class BraintreePaypalPaymentStrategy implements PaymentStrategy {
...
+        const isMessageContainerAvailable =
+            containerId && Boolean(document.getElementById(containerId));
+
+        if (this._braintreeHostWindow.paypal && isMessageContainerAvailable) {
+            const state = this._store.getState();
+            const cart = state.cart.getCartOrThrow();
+
+            this._braintreeHostWindow.paypal
+                .Messages({
+                    amount: cart.cartAmount,
```
Comment (animesh1987) [inline packages/core/src/payment/strategies/braintree/braintree-paypal-payment-strategy.ts:293]: Think you can use subtotal from checkout object.

**Comment by animesh1987** [comment]
The changes look good, but there are quite a few changes with only one test change. Can we add more coverage?

File: packages/core/src/payment/strategies/braintree/braintree-paypal-payment-strategy.ts (hunk 3 above)
Comment (bc-nick) [reply packages/core/src/payment/strategies/braintree/braintree-paypal-payment-strategy.ts]: Done

**Comment by bc-nick** [comment]
> The changes look good, but there are quite a few changes with only one test change. Can we add more coverage?

@animesh1987 added one more test for `braintree-payment-processor`. For the strategy i covered all cases: successful initialization, failed initialization and banner rendering

**Review by animesh1987 — APPROVED** [review]

**Pull request merged**